.atomist/handlers/event/IssueIntegrationTests.ts

```diff
@@ -0,0 +1,178 @@
+import { IntegrationTest, MappedParameter, Secrets } from "@atomist/rug/operations/Decorators";
+import { HandlerContext, MappedParameters, Response } from "@atomist/rug/operations/Handlers";
+import * as github from "../utils/GitHub";
+import * as slack from "../utils/Slack";
+
+import {
+    ChainedInstruction,
+    HandlerChain,
+    HandlerChainDescriptor,
+    ResponseChainingCommandHandler,
+} from "@atomist/rugs/operations/HandlerChains";
+
+/**
+ * Test that an issue is rendered in a Slack channel when created on
+ * GitHub
+ *
+ * 1. Asserting that a repo exists
+ * 2. Ensuring it is associated with a slack channel
+ * 3. Creating an issue in it
+ * 4. Confirming that the lifecycle message is rendered in the channel
+ *
+ * Needs 'repo' scoped token to create repo
+ */
+
+const testFlow: HandlerChainDescriptor = {
+    handle: createRepo,
+    onAny: {
+        handle: assocRepo,
+        onSuccess: {
+            handle: createIssue,
+            onSuccess: {
+                handle: assertIssueCreated,
+                onSuccess: {
+                    handle: closeIssue,
+                    onSuccess: {
+                        handle: assertIssueClosed,
+                        onSuccess: {
+                            handle: notifySuccess,
+                        },
+                    },
+                },
+            },
+        },
+    },
+};
+
+@IntegrationTest("Test that an issue is rendered in a Slack channel when created on GitHub")
+@Secrets(github.TOKEN_PATH, slack.TOKEN_PATH)
+class IssueRenderedTest extends ResponseChainingCommandHandler {
+
+    @MappedParameter(MappedParameters.GITHUB_REPO_OWNER)
+    public owner: string;
+
+    public init(ctx: HandlerContext): ChainedInstruction {
+        return new ChainedInstruction(slack.listUsers(), { owner: this.owner });
+    }
+}
+
+export const handlers =
+    new IssueRenderedTest(testFlow)
+        .buildHandlerChain();
+
+// start: response handler logic functions
+
+/**
+ * Extract slack user Id from response and create repo
+ */
+function createRepo(response: Response<any>, params: any): ChainedInstruction {
+    if (response.code !== 200) {
+        throw new Error(`List users failed with ${response.code}`);
```

Comment:
The reason I have no error handlers is so that the integration test fails and the CLI extension ensures that CommandException is thrown so that we get a non-zero exit code from the CLI, so that I can detect and report those failures to logzio from the integration-testing-service.